### DISCUSSION TITLE
Any replacement for presets ?

### DISCUSSION THREAD

**Discussion by jurelou** [dagster-io/dagster#7371] [Q&A]
Is there any replacement for the good old preset API ? 

I'd like to load predefined configurations into dagit.

Thanks !

**Comment by sryza** [comment]
Hi @jurelou - you can put config directly on a job, and it will show up in Dagit: https://docs.dagster.io/guides/dagster/graph_job_op#a-pipeline-with-a-preset.

**Comment by jurelou** [comment]
Thanks for the answer @sryza.
However i still don't understand how i can achieve my goal.

I have a simple op that takes a mandatory configuration. for example:
```
@op(
    config_schema={
        'file_names': Field([str]),
    },
)
def my_op(context, folder: Path) -> Path:
    logger = get_dagster_logger()
```

I would like to have a "preset" configuration, but still want the user to be able to modify the configuration in dagit.

When i set config for a job, it does not reflect on the GUI. For example:

```
@job(
    config={
        "ops": {
            "my_op": {
                "config": {
                    "file_names": ["some_val"],
                }
            },
        }
    }
)
def myjob():
  ...
```

Also, when i try with job config mappings, i'm not able to modify `my_op` configuration from dagit.

**Reply by sryza** [reply]
@jurelou

> I would like to have a "preset" configuration, but still want the user to be able to modify the configuration in dagit.

You should this be able to do this. 

I tried out something similar to your code snippet and found that the config shows up in the launchpad and is editable.  Are you experiencing otherwise?

Code:
```
from dagster import op, job, get_dagster_logger, Field


@op(
    config_schema={
        "file_names": Field([str]),
    },
)
def my_op(context):
    logger = get_dagster_logger()


@job(
    config={
        "ops": {
            "my_op": {
                "config": {
                    "file_names": ["some_val"],
                }
            },
        }
    }
)
def myjob():
    my_op()

```

Dagit:
<img width="1153" alt="image" src="https://user-images.githubusercontent.com/654855/164990914-1a15d27a-a8f7-4456-9e29-c662945f636c.png">

**Comment by jurelou** [comment]
I got it. When i press "new run" it successfully updates. I guess there is a caching mechanism in the browser. (the value is kept even if I restart the whole stack)

Also, is there any way to provide different "preset" or "run configurations"? So the user can select one of the provided config.

**Reply by sryza** [reply]
Ah - makes sense. To provide different preset or run configurations, you would need to make a different job for each. Alternatively, you could use `static_partitioned_config` to make a partitioned job with different config for each partition: https://docs.dagster.io/concepts/partitions-schedules-sensors/partitions#defining-a-job-with-static-partitions.
